helm/resource_release.go

```diff
@@ -541,9 +541,10 @@ func resourceReleaseCreate(ctx context.Context, d *schema.ResourceData, meta int
 			return diag.FromErr(err)
 		}
 
-		debug("%s Release was created but returned an error", logID)
+		debug("%s Release was created but returned an error, rolling back", logID)
 
-		if err := setReleaseAttributes(d, rel, m); err != nil {
+		if err != nil && exists {
```

Comment:
I think this conditional might be superfluous as it's already implicit that `err != nil` because of the conditional this is nested inside, and we have a conditional above this one to handle the `!exists` case.